tensorly/tenalg/proximal.py

```diff
@@ -92,3 +95,159 @@ def procrustes(matrix):
+        If set to None, the problem is solved unconstrained.
+        Default: None
+    nonzero_rows: boolean
+        True if the lines of the V matrix can't be zero,
+        False if they can be zero
+        Default: False
+
+    Returns
+    -------
+    V: array
+        a r-by-n nonnegative matrix \approx argmin_{V >= 0} ||M-UV||_F^2
+    rec_error: float
+        number of loops authorized by the error stop criterion
+    iteration: integer
+        final number of update iteration performed
+    rho: float
+        number of loops authorized by the time stop criterion
+
+    References
+    ----------
+    [1]: N. Gillis and F. Glineur, Accelerated Multiplicative Updates and
+    Hierarchical ALS Algorithms for Nonnegative Matrix Factorization,
+    Neural Computation 24 (4): 1085-1105, 2012.
+
+    [2] J. Eggert, and E. Korner. "Sparse coding and NMF."
+    2004 IEEE International Joint Conference on Neural Networks
+    (IEEE Cat. No. 04CH37541). Vol. 4. IEEE, 2004.
+
+    """
+
+    rank, n_col_M = tl.shape(UtM)
+    if V is None:  # checks if V is empty
+        V = tl.solve(UtU, UtM)
+
+        V=tl.clip(V,a_min=0,a_max=None)
+        # Scaling
+        scale = tl.sum(UtM * V) / tl.sum(
+            UtU * tl.dot(V, tl.transpose(V)))
+        V = tl.dot(scale, V)
+    if exact:
+        n_iter_max=5000
+        tol = 10e-12
+
+    for iteration in range(n_iter_max):
+        rec_error= 0
+        rec_error0=0
+        for k in range(rank):
+
+            if UtU[k, k]:
+                if sparsity_coefficient is not None: # Modifying the objective function for sparsification
+
+                    deltaV = tl.max([(UtM[k, :] - UtU[k, :] @ V - sparsity_coefficient * tl.ones(n_col_M)) / UtU[k, k],
+                                        -V[k, :]],axis=0)
+                    tl.index_update(V, tl.index[k, :], V[k, :] + deltaV)
+
+                else:  # without sparsity
+
+                    deltaV = tl.max([(UtM[k, :] - tl.dot(UtU[k, :], V)) / UtU[k, k],
+                                         -V[k, :]], axis=0)
+                    V=tl.index_update(V,tl.index[k, :],V[k, :] + deltaV)
+
+                rec_error = rec_error + tl.dot(deltaV, tl.transpose(deltaV))
+
+                # Safety procedure, if columns aren't allow to be zero
+                if nonzero_rows and tl.all(V[k, :] == 0):
+                    V[k, :] = 1e-16 * tl.max(V)
+
+            elif nonzero_rows:
+                raise ValueError("Column " + str(k) + " of U is zero with nonzero condition")
+
+            if normalize:
+                norm = tl.norm(V[k,:])
+                if norm != 0:
+                    V[k,:] /= norm
+                else:
+                    sqrt_n = 1/n_col_M ** (1/2)
+                    V[k,:] = [sqrt_n for i in range(n_col_M)]
+        if iteration == 1:
+            rec_error0 = rec_error
+
+        rho_up=tl.shape(V)[0]*tl.shape(V)[1]+tl.shape(V)[1]*rank
```

Comment:
Could we find a better name for rho*?

Reply:
New names are nominator, denominator and cvg_criterion.